internal/providers/nvidiabluefield/nvidiabluefield.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2025 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package nvidiabluefield
+
+import (
+	"bytes"
+	"os"
+
+	"github.com/coreos/ignition/v2/config/v3_6_experimental/types"
+	"github.com/coreos/ignition/v2/internal/platform"
+	"github.com/coreos/ignition/v2/internal/providers/util"
+	"github.com/coreos/ignition/v2/internal/resource"
+
+	"github.com/coreos/vcontext/report"
+)
+
+var bootfifoPaths = []string{
+	"/sys/bus/platform/devices/MLNXBF04:00/bootfifo",
+}
+
+func init() {
+	platform.Register(platform.Provider{
+		Name:  "nvidiabluefield",
+		Fetch: fetchConfig,
+	})
+}
+
+func fetchConfig(f *resource.Fetcher) (cfg types.Config, rpt report.Report, err error) {
+	for _, bootfifoPath := range bootfifoPaths {
+		raw, err := os.ReadFile(bootfifoPath)
+
+		if os.IsNotExist(err) {
+			continue
+		} else if err != nil {
+			f.Logger.Err("Could not read bootfifo at %s: %v", bootfifoPath, err)
+			continue
+		}
+
+		if len(raw) == 0 {
+			continue
+		}
+
+		data := bytes.Trim(raw, "\x00")
+		cfg, rpt, err = util.ParseConfig(f.Logger, data)
+		return cfg, rpt, err
+	}
+
+	f.Logger.Info("No config found in any of the NVIDIA BlueField bootfifo interfaces")
+	cfg, rpt, err = util.ParseConfig(f.Logger, []byte{})
+	return
+}
```

Comment:
`gofmt` doesn't like this no newline thing. Can you run `gofmt -w` on this file?